system/Database/Query.php

```diff
@@ -275,11 +275,13 @@ protected function compileBinds()
     {
         $sql = $this->finalQueryString;
 
-        $hasNamedBinds = preg_match('/:((?!=).+):/', $sql) === 1;
+        $hasBinds      = strpos($sql, $this->bindMarker) !== false;
+        $hasNamedBinds = ! $hasBinds
+            && preg_match('/:(?!=).+:/', $sql) === 1;
```

Comment:
Could you add `@see https://regex101.com` link for the regex update, it  can be reference in the future

Reply:
@samsonasik I updated the link.